### ISSUE TITLE
force replacement for azuredevops_serviceendpoint_azurerm with 0.4.0

### ISSUE THREAD

**Issue by juicybaba** [microsoft/terraform-provider-azuredevops#735]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azuredevops_serviceendpoint_azurerm`

### Terraform Configuration Files

```hcl
resource "azuredevops_serviceendpoint_azurerm" "default" {
  project_id                = data.azuredevops_project.default.project_id
  service_endpoint_name     = format("arm-%s-%s", var.project.name, var.project.environment.name)
  azurerm_spn_tenantid      = data.azurerm_client_config.default.tenant_id
  azurerm_subscription_id   = data.azurerm_subscription.default.subscription_id
  azurerm_subscription_name = data.azurerm_subscription.default.display_name
  credentials {
    serviceprincipalkey = var.secret.azdo.client.secret
    serviceprincipalid  = var.project.azdo.client.id
  }
}
```

### Expected Behavior

With `0.4.0`, nothing should happen as the resource is there already (this is the behavior with `0.3.0`), or maybe inject `environment` property into the state.

### Actual Behavior

the resource will be replaced due to the new and optional `environment` property. **Is it an expected behavior**? Because it is forcing all `azuredevops_serviceendpoint_azurerm` to be recreated and users have to grant their azdo pipeline permission to all `azuredevops_serviceendpoint_azurerm` again.
![image](https://user-images.githubusercontent.com/79539850/226701099-0dba7cb2-194e-4630-841c-3dc382e1dac3.png)

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by waltervos** [comment]
This should really take the default (AzureCloud) into account. We lost the approvals on our service connections because of this.

Edit: I mean, if the environment is not specified it should be disregarded.

**Comment by xuzhang3** [comment]
This is a breaking changes. An internal `StateUpgradeData` should manege this changes.

**Comment by juicybaba** [comment]
> This is a breaking changes. An internal `StateUpgradeData` should manege this changes.

@xuzhang3 This `environment` change will become a non-breaking change in next release, correct?

**Comment by GeoffCraigVRU** [comment]
When will a new release come out with this fix?

**Comment by xuzhang3** [comment]
@GeoffCraigVRU  v0.5.0 has been released, you can upgrade to v0.5.0
